uint64[] prev_features
---
uint64[] features
int64 score